app/controllers/helpers/AddressJourneyCachingHelper.scala

```diff
@@ -37,6 +37,9 @@ trait AddressJourneyCachingHelper { this: AddressController =>
     implicit hc: HeaderCarrier): Future[CacheMap] =
     sessionCache.cache(s"${typ}SelectedAddressRecord", addressRecord)
 
+  def cacheSelectedRecordSet(typ: AddrType, recordSet: RecordSet)(implicit hc: HeaderCarrier): Future[CacheMap] =
```

Comment:
Refactor this file to use typed identifiers instead of strings, this will allow you to write a more generic function instead of creating a new function every time something needs added to the cache. The following would work
```
trait TypedIdentifier[A] {
  val id: String
}

abstract class AddrTypeId[A](partialId: String) extends TypedIdentifier[A] {
  val typ: AddrType
  val id: String = s"$typ$partialId"
}

case class RecordSetId(typ: AddrType) extends AddrTypeId[RecordSet]("SelectedRecordSet")

case object AddressPageVisitedId extends TypedIdentifier[AddressPageVisitedDto] {
  val id: String = "addressPageVisitedDto"
}
```
This would allow you to create a generic function for caching data like so:
```
def addToCache[A](id: TypedIdentifier[A], record: A)(implicit hc: HeaderCarrier): Future[CacheMap] =
  sessionCache.cache(id.id, record)
```